master/lustrefactory.py

```diff
@@ -56,6 +62,65 @@ def buildzfsCommand(props):
     return args
 
 @util.renderer
+def configureCmd(props):
+    args = ["./configure"]
+    style = props.getProperty('buildstyle')
+
+    if style == "deb" or style == "rpm":
```

Comment:
Do you mean multiple factories which have the slight variation on steps? If so, I prefer this method because all builders follow the same general flow of download tarball, configure, make, maybe rpmbuild (for srpm), and cleanup. Properties are intended to capture and handle these slight variations from my understanding. I'd find it rather confusing and too verbose to have 3 or 4 factories for all the slight variations on the same process when all of them mostly follow the same general steps.